server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
                 updates['source'] = desk['source'] if desk and desk.get('source', '') \
                     else DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES
 
+            updates['pubstatus'] = 'canceled' if self.publish_type == 'killed' else 'usable'
```

Comment:
I know, that's why I said enum _-ish_. ;) `collections.namedtuple`, for instance, is a good candidate, too.